libtelemetry/src/main/java/com/mapbox/android/telemetry/LocationJobService.java

```diff
@@ -0,0 +1,217 @@
+package com.mapbox.android.telemetry;
+
+import android.annotation.SuppressLint;
+import android.app.job.JobInfo;
+import android.app.job.JobParameters;
+import android.app.job.JobScheduler;
+import android.app.job.JobService;
+import android.content.ComponentName;
+import android.content.Context;
+import android.location.Criteria;
+import android.location.Location;
+import android.location.LocationListener;
+import android.location.LocationManager;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.CountDownTimer;
+import android.os.PersistableBundle;
+import android.support.annotation.RequiresApi;
+import android.util.Log;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.Response;
+
+@RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+public class LocationJobService extends JobService implements LocationListener, Callback {
```

Comment:
this requires min sdk version of 21, how are we handling this so we don't drop support for lower versions?